Release 3.4 ⚡🚀

nachocodoner opened the pull request:
Includes as the major change: [Meteor-Rspack integration⚡🚀](https://github.com/meteor/meteor/pull/13910).

More info in the forums: https://forums.meteor.com/t/new-3-4-beta-12-release-faster-builds-smaller-bundles-and-modern-setups-with-the-rspack-integration/64124

Please direct any reviews to that PR, not this release PR.

## Plan

Plan towards official release

- [x] **Launch first phase: 3.4-beta.8** ✅
    This beta delivers the first implementation of Meteor-Rspack integration, supporting all skeleton types and covering long-requested features and issues. It focuses only on Rspack changes, leaving other features for later betas. The goal is to keep changes isolated and avoid issue reports that could shift priority away from the modern bundler.
    **Focus:** *Stability*

- [x] **Launch second phase: 3.4-beta.14** ✅
    This beta expands the Meteor-Rspack integration to cover more project setups and modern configs based on reported needs. It is the chance to add configs for projects relying on Meteor-specific behaviors that lack a direct Rspack alternative. It will also review real-world projects using Meteor build compilers and specifics, and provide modern or fallback configs if any.
    **Focus:** *Extensibility*

- [x] **Launch third phase: 3.4-beta.14** ✅
    This beta includes fixes and new features for the Meteor-Rspack integration, plus other 3.4 additions like [dev-only packages](https://github.com/meteor/meteor/pull/13797) and [long-requested contributions that are ready to merge](https://github.com/meteor/meteor/milestone/122).
    **Focus:** *Contributions*

- [x] **Launch RC phase: 3.4-rc.4** ✅
    This RC will bring a final round of stability fixes. It will be the version closest to the official release, and if stable, it will become the final 3.4 release.
   **Focus:** *Official*

italojs commented:
For the modern-tests folder, what do you think we generate those projects at runtime? I see those applications can be outdated over time and it could be a problem for the tests

nachocodoner commented:
I'd rather get the reviews on https://github.com/meteor/meteor/pull/13910.

> For the modern-tests folder, what do you think we generate those projects at runtime?

Answering your question. We have two kind of tests: skeleton and modern app tests. [Skeleton tests](https://github.com/meteor/meteor/blob/modern-bundler-integration/tools/modern-tests/skeleton.test.js) are generated at runtime using `meteor create`. The modern app tests (eg. [react-router.test.js](https://github.com/meteor/meteor/blob/modern-bundler-integration/tools/modern-tests/react-router.test.js)) uses the apps statically described at [modern-tests/apps](https://github.com/meteor/meteor/tree/modern-bundler-integration/tools/modern-tests/apps) to try configs not covered by skeletons, like dynamic imports, ESM, styling, Babel, swcrc configs, and more advanced setups, also those that will come in the future. [As in the legacy suite](https://github.com/meteor/meteor/tree/modern-bundler-integration/tools/tests/apps), we preconfigure some apps for testing. I think this is better than trying to program every possible config and changing of the files to manage them at runtime.

>  I see those applications can be outdated over time and it could be a problem for the tests

Which problems do you see? We should keep those apps compatible in how they’re configured as tools evolve. If a test fails, that signals an issue to check and fix. It may reveal an incompatibility, and we should keep it as a test case. Similarly with legacy self-tests.

StorytellerCZ commented:
@nachocodoner any plans to add Node 24 or should I prepare update for Node 22?

nachocodoner commented:
> any plans to add Node 24 or should I prepare update for Node 22?

@StorytellerCZ: Node 24 will become LTS in October, so according to the release plan, it makes sense to start preparing for it in this release. Hopefully there are no breaking changes and the transition is smooth, as it has been lately.

boomfly commented:
It seems that meteor coffeescript package will not work with 3.4, or mb i do something wrong?

<img width="1003" height="884" alt="Снимок экрана 2025-09-11 в 16 09 13" src="https://github.com/user-attachments/assets/780a53fe-ccd7-42ed-8be0-d5e3d6978322" />

nachocodoner commented:
@boomfly: have you followed the guide to add support of Coffeescript in your app code when using Rspack?

https://deploy-preview-13910.docs.meteor.com/about/modern-build-stack/rspack-bundler-integration.html#coffeescript

If you add support of Coffeescript using Rspack, the coffeescript meteor package is no longer needed, in favor of the delegation to a modern bundler. I could have managed Rspack/Coffeescript integration automatically as part the integration if I detect you use coffeescript atmosphere package. But I believe the right thing to do is to make this configurable and promote rspack.config.js usage for each project type. However, I may add a new skeleton for coffeescript for a future release (meteor create --coffeescript).

boomfly commented:
Add coffee config and next error. It rewrite absolute import to relative and then says that can't resolve that import:

<img width="1052" height="889" alt="Снимок экрана 2025-09-13 в 18 01 19" src="https://github.com/user-attachments/assets/f2b3fd61-2f12-446a-889c-554664bfb11f" />

UPDATE:

Try different resolve configs, but not success:

```js
import { defineConfig } from '@meteorjs/rspack';

/**
 * Rspack configuration for Meteor projects.
 *
 * Provides typed flags on the `Meteor` object, such as:
 * - `Meteor.isClient` / `Meteor.isServer`
 * - `Meteor.isDevelopment` / `Meteor.isProduction`
 * - …and other flags available
 *
 * Use these flags to adjust your build settings based on environment.
 */
export default defineConfig(Meteor => {
  return {
    module: {
      rules: [
        {
          test: /\.coffee$/i,
          use: [
            {
              loader: 'swc-loader',
              // perserve SWC config in the Meteor project level
              options: Meteor.swcConfigOptions,
            },
            {
              loader: 'coffee-loader',
            },
          ],
        },

        // your other rules
      ]
    },
    resolve: {
      // Allow importing without extensions, in this explicit order
      extensions: ['.coffee', '.js', '.ts'],
      // Resolve `import './dir'` to `./dir/index.*`
      mainFiles: ['index'],
      // Allow extension-less ESM imports like `import './server'`
      fullySpecified: false,
      enforceExtension: false,
      preferAbsolute: true
    }
  };
});
```

nachocodoner commented:
@boomfly: https://github.com/meteor/meteor/tree/release-3.4/tools/modern-tests/apps/coffeescript

This is a sample Coffeescript app used for testing. Could you try to replicate it?

If that doesn’t work, please avoid using relative imports. I want to check if that might be the issue so I can reproduce and fix it later.

boomfly commented:
Can reproduce resolve issue: https://github.com/boomfly/meteor-3.4-rspack-resolve-issue

<img width="1513" height="996" alt="Снимок экрана 2025-09-15 в 20 37 14" src="https://github.com/user-attachments/assets/d74aaddc-0c17-44a0-a519-a96c96b10ebe" />

nachocodoner commented:
@boomfly: Could you check if those issues you reported was solved in latest 3.4-beta.10 version?

boomfly commented:
> @boomfly: Could you check if those issues you reported was solved in latest 3.4-beta.10 version?

Yes, import issue resolved in beta.9 version.

Now server build and start, but client has strange error:

<img width="1359" height="1019" alt="Снимок экрана 2025-09-24 в 13 50 16" src="https://github.com/user-attachments/assets/6f152db4-1521-4bb3-835f-f083195e0b2d" />

nachocodoner commented:
> Yes, import issue resolved in beta.9 version.

I mentioned beta.10, but I released a quick fix on beta.11. Does this happen to you with beta.11? Also after `meteor reset`?

I was able to run [your repository without issues.](https://github.com/boomfly/meteor-3.4-rspack-resolve-issue)

boomfly commented:
It seems that project can run with beta.10. Solve issue with i18n undefined, that var imported at top of file: 

```js
import i18n from '/imports/startup/client/i18n'
```

and rspack ingnores other assignmets to that var in that file. I've removed that line and project started.

Is there a way to disable errors modal in DOM and show errors only in console? Cause for now firebase messaging is working, but same time throws error which not crash app.

<img width="1918" height="980" alt="Снимок экрана 2025-09-24 в 17 51 58" src="https://github.com/user-attachments/assets/3b7e6e5f-05f9-43c8-82eb-7df1efef73e3" />

nachocodoner commented:
> Is there a way to disable errors modal in DOM and show errors only in console?

@boomfly: Yes. You should add the next config in your rspack.config.js file.

``` js
devServer: {
  client: {
    overlay: false, // disable the error modal overlay
  },
},
```

Btw, do you have still any problem in your end you would like to share [in your repository](https://github.com/boomfly/meteor-3.4-rspack-resolve-issue?rgh-link-date=2025-09-24T09%3A01%3A43Z)?

harryadel commented:
I'm attempting to do a migration of 2.x application 3.4 and I'll try to share any findings I've.

One of things I noticed right away is that the app doesn't crash after removing [`compat:graphql`](https://github.com/meteor-compat/meteor-graphql) 

rspack config contains this line though:
```
module: {
      rules: [
        {
          test: /\.gql$/,
          use: [
            {
              loader: 'graphql-tag/loader'
            }
          ]
        }
      ]
    },
```

I thought both are needed and only accidentally removed the meteor package and was surprised that the app didn't crash. Can you verify this @nachocodoner

nachocodoner commented:
@harryadel: If you keep the Rspack configuration, you don’t need the Meteor build plugin package anymore.

Since the app code is now handled by Rspack, it’s the only one responsible for processing files like .gpl. The Meteor package might still work in some cases, but it’s not guaranteed in all scenarios. If you need to process the .gpl files inside a Meteor Atmosphere package, you’ll likely need the meteor build-plugin-compat:graphql. However, it’s not required for your app’s direct code. [I mentioned replacing build plugins in the new docs](https://deploy-preview-13910.docs-online.meteor.com/about/modern-build-stack/rspack-bundler-integration.html#replace-build-plugins), but I forgot to suggest removing Meteor build packages once the handling is fully delegated to Rspack. Even by removing the Meteor package, the app's gpl will be still processed by Rspack, that is the reason for they to still work.

I adapted the [Apollo skeleton](https://github.com/meteor/meteor/tree/modern-bundler-integration/tools/static-assets/skel-apollo) to use a config that ensures `.gql` files are detected (via @graphql-tools/webpack-loader). I'm not fully sure if I used the most stable build plugins from Rspack/Webpack, maybe your approach with `graphql-tag/loader` is more appropriate today. The app worked fine anyway. If you know of a more suitable or well-maintained GraphQL setup, please let me know so I can adjust the examples accordingly.

harryadel commented:
Hello Nacho, I was too quick to dive straight into action. Let me preface this first by saying how amazing this release would be and how grateful I'm to your contributions. This PR came in clutch to save the day!

The reason why this migration is taking place is that the current stakeholders were growing frustrated with the project build time and seriously considering switch over to Nextjs or something else since they already don't use Meteor APIs (Methods, Pub/sub) rather GraphQL. But considering the speed gains I'm noticing hopefully I'll be able to avert their decision. :crossed_fingers: 

Who knows how many other applications could've been saved if this was implemented sooner. So thank you for your efforts. This work that you've been doing is going to shape Meteor for many more years to come. :)


Now onto the nitty gritty stuff
>Since the app code is now handled by Rspack, it’s the only one responsible for processing files like .gpl. The Meteor package might still work in some cases, but it’s not guaranteed in all scenarios. If you need to process the .gpl files inside a Meteor Atmosphere package, you’ll likely need the meteor build-plugin-compat:graphql. However, it’s not required for your app’s direct code. [I mentioned replacing build plugins in the new docs](https://deploy-preview-13910.docs-online.meteor.com/about/modern-build-stack/rspack-bundler-integration.html#replace-build-plugins), but I forgot to suggest removing Meteor build packages once the handling is fully delegated to Rspack. Even by removing the Meteor package, the app's gpl will be still processed by Rspack, that is the reason for they to still work.


If I understand correctly, then this means I'd only need to use build-plugins if attempting to compile a file within a Meteor package? But other than that Meteor applications offset this problem to Rspack? I also didn't know about these docs, good thing that you mentioned it :+1: 

This is amazing news since it'd lessen the burden on both (the core team and Meteor developers) since it'd not be our problem anymore! 

>I adapted the [Apollo skeleton](https://github.com/meteor/meteor/tree/modern-bundler-integration/tools/static-assets/skel-apollo) to use a config that ensures .gql files are detected (via @graphql-tools/webpack-loader). I'm not fully sure if I used the most stable build plugins from Rspack/Webpack, maybe your approach with graphql-tag/loader is more appropriate today. The app worked fine anyway. If you know of a more suitable or well-maintained GraphQL setup, please let me know so I can adjust the examples accordingly.

Honestly I'm still making adjustments to the application to get it working and also fiddling with rspack configs to reach the best setup. Maybe once I'm done I'll share it.

harryadel commented:
```js
const { defineConfig } = require('@meteorjs/rspack');
// const { RsdoctorRspackPlugin } = require('@rsdoctor/rspack-plugin'); // Disabled - compatibility issues
const rspack = require('@rspack/core');

/**
 * Rspack configuration for Meteor projects.
 *
 * Provides typed flags on the `Meteor` object, such as:
 * - `Meteor.isClient` / `Meteor.isServer`
 * - `Meteor.isDevelopment` / `Meteor.isProduction`
 * - …and other flags available
 *
 * Use these flags to adjust your build settings based on environment.
 */
module.exports = defineConfig(Meteor => {
  return {
    // Disable caching - let Meteor handle it
    cache: false,
    // Only add externals for server-side builds to avoid bundling Node.js modules
    ...(Meteor.isServer && {
      externals: {
        // Node.js built-in modules
        'querystring': 'commonjs querystring',
        'url': 'commonjs url',
        'http': 'commonjs http',
        'https': 'commonjs https',
        'fs': 'commonjs fs',
        'path': 'commonjs path',
        'vimeo': 'commonjs vimeo',
        // Sharp and its dependencies
        'sharp': 'commonjs sharp',
        '@img/sharp-libvips-dev/cplusplus': 'commonjs @img/sharp-libvips-dev/cplusplus',
      },
    }),
    resolve: {
      extensions: ['.tsx', '.ts', '.jsx', '.js', '.json'],
      modules: ['node_modules'],
      symlinks: false, // Disable if not using symlinks for better performance
    },
    module: {
      rules: [
        {
          test: /\.gql$/,
          use: [
            {
              loader: 'graphql-tag/loader',
              options: {
                noDescription: Meteor.isProduction, // Remove descriptions in production for smaller bundle
              }
            }
          ]
        }
      ]
    },
    // Simplified optimization - let Meteor handle most of the chunking
    optimization: {
      // Enable tree shaking for unused exports
      usedExports: true,
      innerGraph: true,
      // Production minification
      minimize: Meteor.isProduction,
      ...(Meteor.isProduction && {
        minimizer: [
          new rspack.SwcJsMinimizerRspackPlugin({
            compress: {
              drop_console: true, // Remove console.logs in production
              passes: 2,
            },
          }),
          new rspack.LightningCssMinimizerRspackPlugin(),
        ],
      }),
    },
    // Performance optimizations
    performance: {
      hints: false, // Disable performance hints in development
    },
    // Enable source maps for better debugging - use 'source-map' instead of 'eval-source-map' for RsDoctor compatibility
    devtool: Meteor.isDevelopment ? 'source-map' : false,
    // Plugins for bundle analysis and optimization
    plugins: [
      // RsDoctor disabled - appears to have compatibility issues with Meteor 3.4-beta.11
      // Meteor.isDevelopment && Meteor.isClient && new RsdoctorRspackPlugin({
      //   linter: {
      //     rules: {
      //       'ecma-version-check': 'off',
      //     },
      //   },
      // }),
      // Compression for production builds (client only)
      Meteor.isProduction && Meteor.isClient && new rspack.CopyRspackPlugin({
        patterns: [],
      }),
    ].filter(Boolean),
    // Native CSS experiments
    experiments: {
      css: true,
    },
  };
});
```

Okidoki, so I had some time to fiddle around with rspack and here're my findings:

- `Multiple assets emit different content to the same filename client-rspack.js` error 

`splitChunks` configuration with multiple cache groups was conflicting with Meteor's internal entry point and chunk management system.

**Original Configuration** (INCOMPATIBLE):
```javascript
splitChunks: {
  chunks: 'all',
  maxInitialRequests: 25,
  maxAsyncRequests: 25,
  cacheGroups: {
    react: { /* ... */ },
    mantine: { /* ... */ },
    apollo: { /* ... */ },
    // ... 8 cache groups total
  }
}
```

- `@rsdoctor/rspack-plugin` package causes runtime errors when used with Meteor 3.4-beta.11:
      - `webpack-dev-server` crashes
      - Package.json resolution errors in RsDoctor core
      - Multiple deprecation warnings
      
- Use `devtool: 'source-map'` instead of `devtool: 'eval-source-map'`

- (`cache: { type: 'filesystem', ... }`) caused validation errors as Meteor expects a boolean value

nachocodoner commented:
Thanks for your time and the report @harryadel 

`splitChunks` is worth a closer look. The current Meteor-Rspack integration only emits chunks to the public folder via dynamic imports. I’ll see if I can make the setup more flexible for splitting.

~About externals: you don’t need them for built-in Node.js modules; with `target: node`, Rspack handles those.~ I think we need to set [`externalsPresets: { node: true }`](https://rspack.rs/config/externals#externalspresets). I’ll set it as the default so there’s no need to manually add each built-in module to externals in any app. For other deps like `sharp` or `vimeo`, marking them as external makes Rspack skip them so Meteor’s compiler resolves them. That’s useful only if Rspack compilation causes issues or there’s a Meteor/Rspack incompatibility; otherwise I wouldn’t externalize.

Regarding cache, did you disable it for performance or memory overflow, or because it didn’t react well to changes? I’m improving the default cache under `experiments.cache` to behave faster and more stable on edits.

Some of your configs are already defaults in Meteor-Rspack, like `experiments.css`, `optimization.usedExports` and others. If you enable the verbose setting below and remove your `rspack.config.js` overrides, you’ll see the default client and server configs:

``` json
{
  "meteor": {
    "modern": {
      "verbose": true
    }
  }
}
```

harryadel commented:
Nacho, I'm jumping here quick to tell you that you've outdone yourself with this release. This PR just keeps on giving. I'm migrating some other project to 3.x and instead of 3.3.2 I decided to to amp it to 3.4 and now thanks to your build changes. We no longer need [`fourseven:scss`](https://github.com/Meteor-Community-Packages/meteor-scss)!

This package has been an immense pain in the ass and I [remember how annoying](https://github.com/Meteor-Community-Packages/meteor-scss/pull/321) it was to migrate it to 3.0. This release frees Meteor developers from having to handle build packages ourselves and Meteor can finally piggyback off of the Nodejs ecosystem shedding lots of old code. Whew what a relief!! 

Ahem another rspack config in case you may need it :sweat_smile: 

```js
const { defineConfig } = require('@meteorjs/rspack');

/**
 * Rspack configuration for Meteor projects.
 *
 * Provides typed flags on the `Meteor` object, such as:
 * - `Meteor.isClient` / `Meteor.isServer`
 * - `Meteor.isDevelopment` / `Meteor.isProduction`
 * - …and other flags available
 *
 * Use these flags to adjust your build settings based on environment.
 */
module.exports = defineConfig(Meteor => {
  return {
    module: {
      rules: [
        {
          test: /\.tsx?$/,
          use: 'builtin:swc-loader',
          options: {
            jsc: {
              parser: {
                syntax: 'typescript',
                tsx: true,
              },
            },
          },
          type: 'javascript/auto',
        },
        {
          test: /\.scss$/,
          use: [
            'style-loader',
            'css-loader',
            'sass-loader',
          ],
          type: 'javascript/auto',
        },
        {
          test: /\.css$/,
          use: [
            'style-loader',
            'css-loader',
          ],
          type: 'javascript/auto',
        },
      ],
    },
    resolve: {
      fallback: {
        https: false,
        crypto: false,
      },
    },
  };
});

```

harryadel commented:
> Regarding cache, did you disable it for performance or memory overflow, or because it didn’t react well to changes? I’m improving the default cache under `experiments.cache` to behave faster and more stable on edits.

I disabled it because the build was getting stuck. Also, regarding cache and what not. Whenever I make a change to rspack or somethign I find it better to clear the old Meteor cache `meteor reset` or else I'd get errors! Should this be handled?

Grubba27 commented:
@harryadel very nice work!!! Could you make a PR with those examples for this branch(release-3.4)?

I wonder, and I'll discuss with Nacho if we can have some "helpers" or something like it, so instead of  passing the rules (or remembering it) we could have some vite-style plugins:


```js
const { sass } = require('@meteorjs/rspack/sass');
    module: {
      rules: [
        {
          test: /\.tsx?$/,
          use: 'builtin:swc-loader',
          options: {
            jsc: {
              parser: {
                syntax: 'typescript',
                tsx: true,
              },
            },
          },
          type: 'javascript/auto',
        },
       ...sass() // this would be the same as writting the rules but we give it a "nice" api around it
      ],
```

nachocodoner commented:
@harryadel: Exactly. Relying more on the broader JS ecosystem will let us focus on what we enjoy most—building with Meteor. That’s one of the main goals behind all of this.

We already mention in the docs the direction toward using Rspack as an alternative for Meteor build plugins:

[Replace Build plugins](https://deploy-preview-13910.docs-online.meteor.com/about/modern-build-stack/rspack-bundler-integration.html#replace-build-plugins)

[CSS, Less and SCSS](https://deploy-preview-13910.docs-online.meteor.com/about/modern-build-stack/rspack-bundler-integration.html#css-less-and-scss)

Feel free to expand these sections if you think more details would help.

@Grubba27:

> so instead of passing the rules (or remembering it) we could have some vite-style plugins:

I’ve actually experimented with that recently. Take a look at this section:

With [Meteor.compileWithMeteor and Meteor.compileWithRspack](https://deploy-preview-13910.docs-online.meteor.com/about/modern-build-stack/rspack-bundler-integration.html#compilewithrspack-and-compilewithmeteor), you already have helper presets to build the config. Instead of remembering every setting, these presets automatically merge the right configuration into the final setup.

``` js
module.exports = defineConfig(Meteor => ({
  // Exclude native modules from the bundle (use Meteor runtime)
  ...(Meteor.isServer ? Meteor.compileWithMeteor(['sharp']) : {}),

  // Force-compile modern or local packages via SWC
  ...Meteor.compileWithRspack(['grubba-rpc']),
}));
```

I recently created another one `Meteor.setCache` to also automatically disable the rspack cache config with just setting it to `false`.

harryadel commented:
@nachocodoner Maybe it's time to update the description with a possible ETA?

nachocodoner committed:
refactor test update logic to remove `isTestModule` check

nachocodoner committed:
refactor test module handling and entry point resolution

nachocodoner committed:
refactor test module handling and entry point resolution

nachocodoner committed:
enhanced `rspack.config.js` to support `testClientEntry` and `testServerEntry` settings for better test entry handling

harryadel committed:
Merge branch 'devel' into feature/check

nachocodoner committed:
refactor bundler to handle devDependencies separately

nachocodoner committed:
clean

nachocodoner committed:
Merge branch 'release-3.4' into dev-only-fixes

nachocodoner committed:
refactored the compileWithRspack `pkgDir` logic to handle module resolution more reliably

nachocodoner committed:
Merge pull request #14009 from meteor/dev-only-fixes

Dev only and devDepends atmosphere package fixes

nachocodoner committed:
update BUNDLE_VERSION to 22.21.1.3

nachocodoner committed:
re-run checks

nachocodoner committed:
Meteor version to 3.4-beta.14 :comet:

harryadel committed:
[accounts-password] Fix tests

harryadel committed:
[docs] Add NonEmptyString

nachocodoner committed:
fix async handling in link insertion test

nachocodoner committed:
add Modern Tests workflow for PR-triggered CI

nachocodoner committed:
update test dependencies installation in CI workflow

nachocodoner committed:
refine SWC compiler config for legacy targets

nachocodoner committed:
add legacy build configuration for SWC transpile and minification to fix issues on legacy bundles

nachocodoner committed:
better name

nachocodoner committed:
process rspack outputs for legacy web to make it compatible

nachocodoner committed:
refactor globalThis polyfill to improve legacy support

nachocodoner committed:
fix test after removal of legacy config depending on babel modern minimal versions

nachocodoner committed:
add support for deps installation when Meteor 'update' command and improve messaging

nachocodoner committed:
enhance `compileWithRspack` to support custom options

nachocodoner committed:
add `extendConfig` method to enhance Rspack configuration

nachocodoner committed:
add extendConfig flag and options for compileWithRspack

nachocodoner committed:
Merge pull request #12852 from harryadel/feature/check

Add NonEmptyString to check package

nachocodoner committed:
Merge pull request #14020 from meteor/deps-setup-auto-update

[Rspack] Install tools dependencies as part of `meteor update` and improve messaging

nachocodoner committed:
Merge pull request #14016 from meteor/legacy-fixes-new-bundler

Legacy bundle fixes for modern build stack

nachocodoner committed:
Upgrade @meteorjs/rspack to version 0.2.50

ToyboxZach commented on packages/accounts-base/accounts-base.d.ts:
```diff
@@ -387,8 +387,7 @@ export namespace Accounts {
     user: Meteor.User,
     password: Password
-  ): Promise<{ userId: string; error?: any }>
-}
```
This "}" got deleted, causing type checking to fail

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
add missing closing brace in TypeScript definition file

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

nachocodoner committed:
set environment variables for test polling in CI

nachocodoner committed:
add Meteor preparation step in CI workflow

nachocodoner committed:
add dependency caching to CI workflow

nachocodoner committed:
update cache key to include 'meteor' in hash computation

nachocodoner committed:
use meteor binary for npm installs in test helpers

nachocodoner committed:
disable `fail-fast` in GitHub Actions workflow temporary

nachocodoner committed:
update DEFAULT_RSPACK_VERSION to 1.6.5

nachocodoner committed:
update DEFAULT_RSPACK_VERSION to 1.6.5

nachocodoner committed:
run temporary only babel and react

nachocodoner committed:
only run one test

nachocodoner committed:
enable React tests and set 'fail-fast' to false in CI

nachocodoner committed:
update wait durations to be dynamic based on CI environment

nachocodoner committed:
only run react app tests

nachocodoner committed:
enable Babel and Monorepo tests in CI workflow

nachocodoner committed:
add support for extra entry for the client module in eager test file generation

nachocodoner committed:
add support for extra entry for the client module in eager test file generation

nachocodoner committed:
upgrade @meteorjs/rspack to version 0.2.51

nachocodoner committed:
update option name from 'regExp' to 'include' in test.js

nachocodoner committed:
update @meteorjs/rspack to version 0.2.52

nachocodoner committed:
refactor handling of `extraEntry` regular expression

nachocodoner committed:
upgrade @meteorjs/rspack to version 0.2.53

nachocodoner committed:
add Puppeteer-based browser integration tests

nachocodoner committed:
update selector in browser test to target #react-target

nachocodoner committed:
refactor test file content creation logic

nachocodoner committed:
update @meteorjs/rspack to version 0.2.54

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
re-run checks

nachocodoner committed:
re-run checks

nachocodoner committed:
Merge pull request #14024 from meteor/browser-test-app-rspack

[Rspack] Support browser tests in full app mode

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

# Conflicts:
#	npm-packages/meteor-rspack/lib/test.js

nachocodoner committed:
re-enable some tests categories to ensure stability

nachocodoner committed:
update cache paths in GitHub Actions workflow

nachocodoner committed:
Update modern-tests.yml

nachocodoner committed:
Update @meteorjs/swc-core to version 1.15.3

nachocodoner committed:
refactor dependency installation to separate dev and regular deps

nachocodoner committed:
Update BUNDLE_VERSION to 22.21.1.4

nachocodoner committed:
re-run checks

nachocodoner committed:
Meteor version to 3.4-rc.0 :comet:

nachocodoner committed:
update @rspack/cli and @rspack/core to version 1.6.5

nachocodoner committed:
upgrade Playwright to version 1.57.0

nachocodoner committed:
Meteor version to 3.4-rc.1 :comet:

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

nachocodoner committed:
improve test reliability with retry logic in workflows

nachocodoner committed:
add retries and enhance stability for modern tests

Grubba27 committed:
FEATURE: Add deferrable functions to meteor.d.ts

nachocodoner committed:
add retries and enhance stability for modern tests

nachocodoner committed:
add retry wait time to modern-tests workflow

nachocodoner committed:
set Jest retries conditionally for CI environment

nachocodoner committed:
update modern-tests caching and remove Playwright installation

nachocodoner committed:
enable Blaze and React in modern-tests workflow

nachocodoner committed:
fix browser test args by adding --disable-dev-shm-usage

nachocodoner committed:
clean package-lock for failing test in full blaze example

nachocodoner committed:
simplify test workflow by removing unused environment variables

nachocodoner committed:
refactor ReactRouter references to R.Router

nachocodoner committed:
enable composite builds in TypeScript configuration

nachocodoner committed:
add Angular to test matrix in GitHub Actions workflow

nachocodoner committed:
skip Typescript skeleton test and update rspack CI config

nachocodoner committed:
remove retry_wait_seconds from modern-tests workflow

nachocodoner committed:
add concurrency control to modern-tests workflow

nachocodoner committed:
add file exclusion patterns to TypeScript checker config

nachocodoner committed:
add retry wait time to test workflow

nachocodoner committed:
simplify TypeScript test exclusions in rspack.config.ts

nachocodoner committed:
refactor skeleton test structure for consistency

nachocodoner committed:
add wait step before Rspack package setup in tests

nachocodoner committed:
remove resource-intensive TypeScript checks in CI builds

nachocodoner committed:
disable GPU usage in browser tests configuration

nachocodoner committed:
set NODE_OPTIONS to increase memory limit in CI workflow

nachocodoner committed:
remove redundant Chrome flags from browser test config

nachocodoner committed:
adjust browser test configuration for stability

nachocodoner committed:
simplify test file content generation logic

nachocodoner committed:
update step name in workflow to generalize dependency install

Grubba27 committed:
Merge pull request #14029 from meteor/feature/add-deferrable-functions-to-meteor.d.ts

FEATURE: Add deferrable functions to meteor.d.ts

nachocodoner committed:
update workflow name and concurrency group for clarity

nachocodoner committed:
refactor killProcessByPort to handle multiple ports

Grubba27 committed:
FEATURE: Remove `import React from 'react';` from skels

With our SWC changes, we can both remove this import line and add an example of swc configuration

Grubba27 committed:
Merge pull request #14041 from meteor/feature/remove-react-import

FEATURE: Remove `import React from 'react';` from skels

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

nachocodoner committed:
rename and update workflow for E2E tests

nachocodoner committed:
update test descriptions to cover pending skeleton test cases

nachocodoner committed:
re-run checks

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

nachocodoner committed:
simplify cache key by removing package-lock.json hash

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
Merge branch 'release-3.4' into modern-tests-ci-check

nachocodoner committed:
add step to handle NPM dependency updates via `meteor update`

nachocodoner committed:
Merge pull request #14010 from meteor/modern-tests-ci-check

E2E Tests CI check

nachocodoner committed:
Merge branch 'release-3.4' into update-npm-cmd

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
simplify Meteor preparation step in Windows CI workflow

nachocodoner committed:
Merge remote-tracking branch 'origin/release-3.4' into release-3.4

nachocodoner committed:
Revert "simplify Meteor preparation step in Windows CI workflow"

This reverts commit 0c4422a7469bcca6680361f217e8ef12487339f4.

nachocodoner committed:
add submodule reset step to Windows CI workflow

nachocodoner committed:
Merge branch 'release-3.4' into update-npm-cmd

nachocodoner committed:
add AssetExternalsPlugin to handle asset externalization

nachocodoner committed:
add `disablePlugins` helper to filter Rspack plugins by matchers

nachocodoner committed:
use original file names for public assets in Rspack config

nachocodoner committed:
support assets automatic resolution to public folder and test coverage several scenarios

nachocodoner committed:
add test coverage CustomConsoleLogPlugin with support for plugin disabling

nachocodoner committed:
ensure output.publicPath can be overriden

Grubba27 committed:
FEATURE: Update React skeleton to use React Router

We long discussed having a more modernized base React skeleton that uses
a router and has _some_ basic navigation and styling. Check below a video and some images of the
new React skeleton in action.

nachocodoner committed:
Update 3.4.0.md

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
Merge branch 'release-3.4' into update-npm-cmd

nachocodoner committed:
refactor plugin filtering logic in meteorRspackHelpers

nachocodoner committed:
upgrade @meteorjs/rspack to version 0.3.53

nachocodoner committed:
Merge remote-tracking branch 'origin/ext-assets-style-files' into ext-assets-style-files

# Conflicts:
#	v3-docs/docs/generators/changelog/versions/3.4.0.md

nachocodoner committed:
refactor background image URL check in test logic

nachocodoner committed:
update @meteorjs/rspack to version 0.3.54 across packages

nachocodoner committed:
Merge branch 'release-3.4' into ext-assets-style-files

nachocodoner committed:
re-run checks

nachocodoner committed:
Merge branch 'release-3.4' into ext-assets-style-files

nachocodoner committed:
add rspack comment to Meteor skeletons and test apps

nachocodoner committed:
add validation for Meteor entry points in Rspack plugin

nachocodoner committed:
increase retry wait time for e2e test workflow

nachocodoner committed:
Extend timeout for e2e test workflow tasks

nachocodoner committed:
extend timeout for e2e test workflow tasks

nachocodoner committed:
Merge pull request #14054 from meteor/ext-assets-style-files

Support assets to resolve to public folders within style files and test coverage several scenarios

nachocodoner committed:
Merge branch 'release-3.4' into update-npm-cmd

Grubba27 committed:
DEV: use logo as svg, add svgr/webpack and use `""`

nachocodoner committed:
Merge pull request #14049 from meteor/update-npm-cmd

Support `meteor update --npm` to pre-bump rspack dependencies on updates

nachocodoner committed:
Update test description for ChakraUI Skeleton

nachocodoner committed:
Update BUNDLE_VERSION to 22.21.1.5

nachocodoner committed:
re-run checks

Grubba27 committed:
Merge branch 'release-3.4' into feature/update-react-skeleton-to-use-react-router

nachocodoner committed:
Meteor version to 3.4-rc.2 :comet:

StorytellerCZ commented:
There will be a Node security release tomorrow: https://nodejs.org/en/blog/vulnerability/december-2025-security-releases

I suggest we incorporate that release and then release.

nachocodoner commented:
@StorytellerCZ: Good point. I can either release an RC.3 later, or just include it in Meteor 3.4 official in January (expected to be announced then). Could you share the Node PR when available?

Also, I already released RC.2 to test a new functionality: https://github.com/meteor/meteor/pull/14049. This lets me move testing forward. That PR might still need a new release if something is wrong in the tool.

Adding options to `meteor update` is hard to test, but I changed the global Meteor version directly and it seemed to work. I still need to double-check it with the final published version.

StorytellerCZ commented:
@nachocodoner I'll get the PR for updating Node ready once I see that the release happened. Depending on the security issue we might need to do another 3.3 patch release if 3.4 will be released in January.

UPDATE: The security release for Node has been postponed till January 8th: https://nodejs.org/en/blog/vulnerability/december-2025-security-releases

Grubba27 committed:
UX: update styling for skeleton

Grubba27 committed:
DEV: remove router

Grubba27 committed:
Merge branch 'release-3.4' into feature/update-react-skeleton-to-use-react-router

Grubba27 committed:
Merge pull request #14056 from meteor/feature/update-react-skeleton-to-use-react-router

FEATURE: Update React skeleton styling and support SVGs

nachocodoner committed:
re-run checks

nachocodoner committed:
refactor bodyStyles handling to fix react skeleton test and adjust font loading

nachocodoner committed:
added a lazy compilation configuration to disable it due to Meteor's compatibility constraints

nachocodoner committed:
update Rspack dependencies to version 1.7.1

nachocodoner committed:
Merge pull request #14076 from meteor/fix-react-e2e

Fix React Skeleton and E2E test

nachocodoner committed:
Merge branch 'release-3.4' into support-rspack-1.7.0

nachocodoner committed:
add warning for potential issues with lazyCompilation

nachocodoner committed:
Merge pull request #14077 from meteor/support-rspack-1.7.0

Support Rspack 1.7.x (lazyCompilation disabled by default)

nachocodoner committed:
update BUNDLE_VERSION to 22.21.1.6

nachocodoner committed:
re-run checks

nachocodoner committed:
Meteor version to 3.4-rc.3 :comet:

nachocodoner committed:
add Docker setup instructions for Meteor with Rspack

StorytellerCZ committed:
Node 22.22.0

https://github.com/nodejs/node/releases/tag/v22.22.0

nachocodoner committed:
remove Rspack cache documentation from integration guide as repeated

nachocodoner committed:
update React tutorial with Rspack and HMR improvements

nachocodoner committed:
update HMR details for Meteor 3.4+ in tutorials

nachocodoner committed:
update React tutorial to include Rspack and Meteor bundler details

nachocodoner committed:
update React tutorial to include Rspack and Meteor bundler details

nachocodoner committed:
update React tutorial to include Rspack and Meteor bundler details

nachocodoner committed:
update Vue tutorial references and adopt Rspack bundler

nachocodoner committed:
update Vue tutorial references and adopt Rspack bundler

nachocodoner committed:
update Vue tutorial references and adopt Rspack bundler

nachocodoner committed:
update Vue tutorial references and adopt Rspack bundler

italojs committed:
Bump dev bundle version to 22.22.0.1

nachocodoner committed:
update Solid tutorial references and adopt Rspack bundler

nachocodoner committed:
update Blaze tutorial references and adopt Rspack bundler

nachocodoner committed:
update Svelte tutorial references and adopt Rspack bundler

nachocodoner committed:
update Svelte tutorial references and adopt Rspack bundler

nachocodoner committed:
update build plugins for packages docs to reflect Rspack limitations in Meteor 3.4+

nachocodoner committed:
add links to Solid, Blaze, and Svelte 5 tutorials in docs

nachocodoner committed:
update Rspack plugin references in documentation

italojs committed:
trigger ci

nachocodoner committed:
Merge pull request #14092 from meteor/dev-bundle-22.22.0.1

Dev bundle 22.22.0.1

nachocodoner committed:
update BUNDLE_VERSION to 22.22.0.2

nachocodoner committed:
re-run checks

nachocodoner committed:
refactor info blocks to add the two bundler version links to one

nachocodoner committed:
refactor info blocks to add the two bundler version links to one

nachocodoner committed:
improve tutorial clarity on Rspack and Meteor bundlers

nachocodoner committed:
Merge branch 'release-3.4' into docs-improve-rspack

nachocodoner committed:
install chromium-headless-shell specifically to ensure e2e tests pass

nachocodoner committed:
increase memory limit for E2E tests in CI workflow

nachocodoner committed:
update install:modern script to adjust Playwright installs

nachocodoner committed:
update install:modern script to adjust Playwright installs

nachocodoner committed:
pin Playwright to version 1.58.0 to ensure consistency on browser installed

nachocodoner committed:
update Playwright to version 1.58.0

nachocodoner committed:
Update e2e-tests.yml

nachocodoner committed:
Merge pull request #14098 from meteor/fix-e2e-playwright-tests

Fix E2E/Playwright tests recently failing

nachocodoner committed:
Merge branch 'release-3.4' into docs-improve-rspack

nachocodoner committed:
update roadmap with new release goals and plans

nachocodoner committed:
update roadmap to clarify Express integration details

nachocodoner committed:
Merge branch 'release-3.4' into 3.4-roadmap

nachocodoner committed:
remove dead links in TODO placeholders

nachocodoner committed:
Merge remote-tracking branch 'origin/3.4-roadmap' into 3.4-roadmap

nachocodoner committed:
update roadmap wording for clarity

nachocodoner committed:
update roadmap wording

nachocodoner committed:
upgrade @meteorjs/rspack to official version 1.0.0

nachocodoner commented:
Releasing the last RC.4 to validate CI releasing procedures before 3.4 official. Right after that, 3.4 official is planned to be released and recommended on January 30th, if everything goes well.

nachocodoner committed:
update BUNDLE_VERSION to 22.22.0.3

nachocodoner committed:
re-run checks

nachocodoner committed:
Meteor version to 3.4-rc.4

nachocodoner committed:
Meteor version to 3.4 :comet:

nachocodoner committed:
improve docker docs section to explain the npm bump issue

nachocodoner committed:
Merge pull request #14082 from meteor/docs-improve-rspack

Review Rspack documentation and tutorials to align with Meteor 3.4 expectations

nachocodoner committed:
Merge branch 'release-3.4' into 3.4-roadmap

italojs committed:
docs(roadmap): expand Change Streams section with phased plan

- Replace placeholder with explanation of leveraging MongoDB Change Streams for real-time updates and efficiency.
- Add Phase 1 (3.5) and Phase 2 (3.5.x) with targets and goals (opt-in initial implementation; configurable implementation + feedback).
- Adjust Next priorities list and clean up related wording.

italojs committed:
docs(roadmap): add Change Streams TODO links; expand Next priorities descriptions (Capacitor, CI/CD, OpenTelemetry, TypeScript, Test toolkit)

italojs committed:
docs(roadmap): clarify Change Streams intro and Phase 1 goal (opt-in, transparent, switch via settings.json/env)

nachocodoner committed:
re-run checks

nachocodoner committed:
Merge remote-tracking branch 'origin/release-3.4' into release-3.4

nachocodoner committed:
document .meteorignore and METEOR_IGNORE

nachocodoner committed:
document memory crashes troubleshotting

nachocodoner committed:
move docker to troubleshotting

nachocodoner committed:
clean

nachocodoner committed:
add link to video on Meteor 3 modern build stack

nachocodoner committed:
update links to Meteor build stack resources

nachocodoner committed:
Merge branch 'release-3.4' into 3.4-roadmap

nachocodoner committed:
remove todos to later cover them

nachocodoner committed:
Merge remote-tracking branch 'origin/3.4-roadmap' into 3.4-roadmap

nachocodoner committed:
update roadmap to reflect Modern Build Stack focus

nachocodoner committed:
update roadmap to reflect Modern Build Stack focus

nachocodoner committed:
Merge pull request #14099 from meteor/3.4-roadmap

Update Roadmap (3.4+)

nachocodoner committed:
fix delimiter for info

nachocodoner committed:
Merge pull request #14100 from meteor/update-docs-3.4

Add missing docs for Meteor 3.4

nachocodoner committed:
update Meteor version to 3.4

nachocodoner commented:
I can't believe it, I'm going to merge this at last! Long live the modernizations to come! 🎉

harryadel commented:
???????????? how is this even getting merged?? this is nothing short of a miracle!
